File: frontend/src/services/userService.js
import api from './api';

const userService = {
  // Отримати всіх користувачів (потрібно додати endpoint на backend)
  getAllUsers: () => api.get('/users'),

  // Отримати користувача за ID
  getUserById: (id) => api.get(`/users/${id}`),

  // Створити нового користувача (використовує /auth/register)
  createUser: (userData) => api.post('/auth/register', userData),

  // Оновити користувача
  updateUser: (id, userData) => api.put(`/users/${id}`, userData),

  // Видалити користувача
  deleteUser: (id) => api.delete(`/users/${id}`),
};

export default userService;

File: backend/src/main/java/com/library/service/UserService.java
package com.library.service;

import com.library.dto.UserDTO;
import com.library.model.User;

import java.util.List;

/**
 * Інтерфейс для роботи з користувачами (Interface Segregation Principle)
 */
public interface UserService {

    UserDTO createUser(UserDTO userDTO);

    UserDTO getUserById(Long id);

    UserDTO getUserByUsername(String username);

    List<UserDTO> getAllUsers();

    UserDTO updateUser(Long id, UserDTO userDTO);

    void deleteUser(Long id);

    boolean existsByUsername(String username);

    boolean existsByEmail(String email);
}
